File: python.Euler/E7.py
def is_primes(x): 
    for i in range(2,x//2):
        if x % i == 0:
            return False
    return True

def main():
    pr = []
    c = 2
    while len(pr) <=10001:
        if is_primes(c)  :
            pr.append(c)
        c += 1
    print(max(pr))

if __name__ == '__main__':
    main()

File: NIM_Euler/nim.Euler/E7.nim
import math,sequtils

proc is_primes(x:int):bool =
  for i in 2..(x div 2):
    if x mod i == 0:
      return false
    return true

proc main =
  var i = 2
  var a:seq[int64] = @[]
  while a.len <= 10001:
    if is_primes(i):
      a.add(i)
    i += 1
  echo a.max

if isMainModule:
  main()
